From 31c008d3f10d4dbd07d5fc3882b434a281973740 Mon Sep 17 00:00:00 2001
Message-Id: <31c008d3f10d4dbd07d5fc3882b434a281973740.1415880041.git.chang-joon.lee@intel.com>
In-Reply-To: <4ea77e6cfc48770ef85ec77ae3cdde5d8482417c.1415880041.git.chang-joon.lee@intel.com>
References: <4ea77e6cfc48770ef85ec77ae3cdde5d8482417c.1415880041.git.chang-joon.lee@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Wed, 12 Nov 2014 11:34:36 +0530
Subject: [PATCH 2/5] SQUASHME! [VPG]: drm/i915: Null check for crtc pointer
 in DPST function

SQUASHME! - This patch should be squashed into the following existing patch:

commit 811fe5ed4bb7ac18f940d89d7441a3d23fadcd6a
Author: daisysun <daisy.sun@intel.com>
Date:   Wed Feb 26 12:06:27 2014 -0800

    MUST_REBASE [VPG]: drm/i915: DPST work on newly attached pipe

Issue: GMINL-3950
Change-Id: I2a8b5425f5c9d99c24b47287292e039aaaa13e51
Signed-off-by: Deepak M <m.deepak@intel.com>
Signed-off-by: Akash Goel <akash.goel@intel.com>
---
 drivers/gpu/drm/i915/i915_dpst.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_dpst.c b/drivers/gpu/drm/i915/i915_dpst.c
index 3aa5861..d0679fa 100644
--- a/drivers/gpu/drm/i915/i915_dpst.c
+++ b/drivers/gpu/drm/i915/i915_dpst.c
@@ -65,6 +65,8 @@ i915_dpst_save_conn_on_edp(struct drm_device *dev)
 		if (i_connector->encoder
 			&& (i_connector->encoder->type == INTEL_OUTPUT_EDP ||
 			i_connector->encoder->type == INTEL_OUTPUT_DSI)) {
+			if (i_connector->encoder->base.crtc == NULL)
+				continue;
 			dev_priv->dpst.connector = i_connector;
 			new_pipe = to_intel_crtc(i_connector->encoder->base.crtc)->pipe;
 			if (new_pipe != dev_priv->dpst.pipe)
-- 
1.7.9.5

